### PR TITLE
Add git commit version info

### LINKED ISSUE #258: Git commit in mosh -v
It would be nice if `mosh{,-client,-server}` built from Git would include the output of `git describe` with the version number.

### PR DESCRIPTION
**Pull request by cgull**
Fixes #258.

### CODE REVIEW

File: scripts/Makefile.am [hunk 1]
```diff
@@ -2,3 +2,9 @@ EXTRA_DIST = wrap-compiler-for-flag-check
 if BUILD_CLIENT
   dist_bin_SCRIPTS = mosh
 endif
+
+mosh:	mosh.pl ../VERSION
+	@sed -e "s/\@VERSION\@/`cat ../VERSION`/" -e "s/\@PACKAGE_STRING\@/@PACKAGE_STRING@/" mosh.pl > mosh
```
Comment (glensc) [inline scripts/Makefile.am]: why is the second replacement `@PACKAGE_STRING@` did you mean `$(PACKAGE_STRING)` ?

Comment (cgull) [inline scripts/Makefile.am]: configure does the substitution when the makefiles are built.  PACKAGE_STRING is set in configure.ac.

Comment (andersk) [inline scripts/Makefile.am]: `$(PACKAGE_STRING)` is still better style. Automake will automatically generate the corresponding assignment `PACKAGE_STRING = @PACKAGE_STRING@`, which allows the variable to be overridden on the `make` command line if desired.

Comment (cgull) [inline scripts/Makefile.am]: > |$(PACKAGE_STRING)| is still better style. Automake will automatically
> generate the corresponding assignment |PACKAGE_STRING =
> @PACKAGE_STRING@|, which allows the variable to be overridden on the
> |make| command line if desired.

But if you do that, then the compile-time override is inconsistent with
the Configure-time definition of |PACKAGE_STRING| in config.h, which is
used for mosh-client and mosh-server.  That can be changed too, of
course, but is there a good reason to have a compile-time knob for this?

**Comment by cgull** [comment]
I should add, I don't know that well what I'm doing here with automake, I cargo-culted this code in from someplace else. If anyone has a use case where compile time configuration for this is useful, I'll be glad to hear it.

**Comment by glensc** [comment]
so, just add another commit changing it to:
```
-e "s/\@PACKAGE_STRING\@/$(PACKAGE_STRING)/" mosh.pl > mosh
```

**Comment by cgull** [comment]
That much is obvious.  Less obvious is why setting PACKAGE_STRING outside of Make's dependency chain is useful.  Have you got a use case?

**Comment by glensc** [comment]
dunno, just guessing, a custom vendor build:

`make PACKAGE_STRING="mosh(patched-by-seagulls)"`

**Comment by cgull** [comment]
I've updated the diff but the scope remains the same-- just the git commit info.  I've gone as far into the bowels of autoconf/automake as I'm going tonight, and still don't know how to make `PACKAGE_STRING` work.  I look forward to a pull request with fully-functional `PACKAGE_STRING` functionality from one of you :)

**Comment by cgull** [comment]
Pulled, plus a slight change to make mosh.pl work without edits.

**Pull request closed without merging**